### ISSUE TITLE
Git gutter not shown in workspace with multiple folders

### ISSUE THREAD

**Issue by anthonyvdotbe** [microsoft/vscode#176738]
Type: <b>Bug</b>

To reproduce:
1. open VS Code
2. open a folder and choose a git repo with changes
3. open a file with changes and notice the gutter is shown
4. add a folder to the workspace and choose a git repo
5. notice the gutter is still visible
6. now close the file and reopen it

Expected behavior: the gutter is shown
Actual behavior: the gutter is not shown

@alexr00 this may be related to #176051. @joyceerhl actually [described](https://github.com/microsoft/vscode/issues/176051#issuecomment-1460604807) pretty much the same issue. However a reload (I assume that is `Developer: Reload Window`) doesn't help in this case.

VS Code version: Code - Insiders 1.77.0-insider (b60db78000bfa99d1b290194ed4651c4ceef02cc, 2023-03-10T05:24:23.805Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10610U CPU @ 1.80GHz (8 x 2304)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.61GB (2.69GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30422396
pythontb:30258533
pythonptprofiler:30281269
vsdfh931cf:30280410
vshan820:30294714
pythondataviewer:30285072
vscod805:30301674
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30404738
cppdebug:30492333
vsclangdf:30492506
c4g48928:30535728
dsvsc012:30540252
pynewext54:30618038
pylantcb52:30590116
pyindex848:30611229
nodejswelcome1:30587009
pyind779:30611226
pythonsymbol12:30651887
showlangstatbar:30659908
pythonb192cf:30661257
funwalk2cf:30682975
pythonms35:30671666
19089226:30680468
7ij38806:30677315

```

</details>

**Comment by alexr00** [comment]
Thanks for the @ mention and for the repro steps!

**Comment by zafodB** [comment]
Also having the same issue. Installing a previous version (`1.75.1`) helped to solve it (temporarily, until it's fixed).

**Comment by alexr00** [comment]
Folk who are seeing this issue: Could you please install the latest insiders build and confirm that you no longer see the issues there?

**Comment by Guyutongxue** [comment]
> Folks who are seeing this issue: Could you please install the latest insiders build and confirm that you no longer see the issues there?

Yes. The issue resolved in the latest insider version.

**Comment by starball5** [comment]
Related Stack Overflow questions:

- [Why aren't SCM diff decorations showing in the gutter in VS Code 1.76.1?](https://stackoverflow.com/q/75715788/11107541) I'd prefer using this one as a duplicate target on Stack Overflow since it's asked much cleaner.
- [Why aren't SCM diff decorations showing in the VS Code gutter in VS Code 1.76.1?](https://stackoverflow.com/q/75709132/11107541)
- [Git Changes indicator does not appear in VSCode](https://stackoverflow.com/q/75715956/11107541)
- [VScode does not highlight the modified line](https://stackoverflow.com/q/75713023/11107541)
- [Vscode stopped highlighting file changes](https://stackoverflow.com/q/75719063/11107541)
- [Visual Studio Code not showing file changes on left side bar](https://stackoverflow.com/q/75721636/11107541)

**Comment by alexr00** [comment]
Thanks for confirming that the issue is fixed in insiders!

**Comment by AnonymousWP** [comment]
Wow, so it was a bug, not an actual symlink-issue as I thought. So happy now, thanks! Context: https://github.com/microsoft/vscode/issues/5970#issuecomment-1470260667

**Comment by galaxytass** [comment]
This issue seems to be fixed in version 1.76.2 for git repositories, however it is not yet fixed for Mercurial (mrcrowl.hg extension).

**Comment by armin-164** [comment]
Im having the same issues, reinstalling helped fix the current file having the issue but any file created after wont show the indicators. Disabling extensions wont work either and Im not sure on how to reproduce the steps

Version: 1.76.2 (user setup)
OS: Windows_NT x64 10.0.19045
